reportportal_client/service.py

```diff
@@ -85,6 +86,27 @@ def _get_messages(data):
     return error_messages
 
 
+def _perform_reliably(request, **kwargs):
+    """Fix for https://github.com/reportportal/client-Python/issues/94.
+
+    Attempt to perform provided HTTP request with kwargs,
+    Ingore KeyError exception first HTTP_REQUEST_RETRY_COUNT times.
+    Return a response or propagate last exception
+        if HTTP_REQUEST_RETRY_COUNT exceeded.
+    """
+    res = None
```

Comment:
Why did not you pick up the requested code block? It contains 4 lines less than yours.

```
    error = None
    for attempt in range(HTTP_REQUEST_RETRY_COUNT):
        try:
            return request(**kwargs)
        except KeyError as err:
            error = err
    raise error
```

Reply:
Agree, thanks